airbyte-cdk/python/airbyte_cdk/sources/declarative/datetime/min_max_datetime.py

```diff
@@ -3,38 +3,41 @@
 #
 
 import datetime as dt
-from typing import Any, Mapping, Optional, Union
+from dataclasses import InitVar, dataclass, field
+from typing import Any, Mapping, Union
 
 from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from dataclasses_jsonschema import JsonSchemaMixin
 
 
-class MinMaxDatetime:
+@dataclass
+class MinMaxDatetime(JsonSchemaMixin):
     """
     Compares the provided date against optional minimum or maximum times. If date is earlier than
     min_date, then min_date is returned. If date is greater than max_date, then max_date is returned.
     If neither, the input date is returned.
+
+    Attributes:
+        datetime (Union[InterpolatedString, str]): InterpolatedString or string representing the datetime in the format specified by `datetime_format`
+        datetime_format (str): Format of the datetime passed as argument
+        min_datetime (Union[InterpolatedString, str]): Represents the minimum allowed datetime value.
+        max_datetime (Union[InterpolatedString, str]): Represents the maximum allowed datetime value.
     """
 
-    def __init__(
-        self,
-        datetime: Union[InterpolatedString, str],
-        datetime_format: str = "",
-        min_datetime: Union[InterpolatedString, str] = "",
-        max_datetime: Union[InterpolatedString, str] = "",
-        **options: Optional[Mapping[str, Any]],
-    ):
-        """
-        :param datetime: InterpolatedString or string representing the datetime in the format specified by `datetime_format`
-        :param datetime_format: Format of the datetime passed as argument
-        :param min_datetime: InterpolatedString or string representing the min datetime
-        :param max_datetime: InterpolatedString or string representing the max datetime
-        :param options: Additional runtime parameters to be used for string interpolation
-        """
-        self._datetime_interpolator = InterpolatedString.create(datetime, options=options)
-        self._datetime_format = datetime_format
+    datetime: Union[InterpolatedString, str]
+    options: InitVar[Mapping[str, Any]]
+    datetime_format: str = ""
+    _datetime_format: str = field(init=False, repr=False, default="")
```

Comment:
This is a unique case where we need the ability to get/set the datetime_format, since we will inherit it from the parent if it is not explicitly defined on the component itself.